source/util/small_vector.h

```diff
@@ -461,15 +462,14 @@ class SmallVector {
   // The number of elements in |small_data_| that have been constructed.
   size_t size_;
 
+  // The actual data used to store the array elements.  It must never be used
+  // directly, but must only be accessed through |small_data_|.
+  alignas(T) std::array<std::byte, sizeof(T) * small_size> buffer;
```

Comment:
Should each element be aligned? Or is that a non-requirement?

```cpp
#include <array>
#include <cstddef>
#include <cstdio>
#include <stdint.h>

template <typename T>
void foo() {
  std::array<T, 2> array;
  void *ptr0 = (void*) &array[0];
  void *ptr1 = (void*) &array[1];
  printf("%p\n", ptr0);
  printf("%p\n", ptr1);
}

template <typename T>
void bar() {
  typename std::aligned_storage<sizeof(T), std::alignment_of<T>::value>::type array[2];
  void *ptr0 = (void*) &array[0];
  void *ptr1 = (void*) &array[1];
  printf("%p\n", ptr0);
  printf("%p\n", ptr1);
}

template <typename T>
void baz() {
  alignas(T) std::array<std::byte, sizeof(T) * 2> array;
  void *ptr0 = (void*) &array[0];
  void *ptr1 = (void*) &array[1];
  printf("%p\n", ptr0);
  printf("%p\n", ptr1);
}

int main() {
  foo<uint32_t>();
  bar<uint32_t>();
  baz<uint32_t>();
  return 0;
}
```

```
[nathan@Cassian ~]$ clang++ -std=c++17 main.cc -o main -O0 && ./main
0x7ffd54d56d00
0x7ffd54d56d04
0x7ffd54d56d00
0x7ffd54d56d04
0x7ffd54d56d00
0x7ffd54d56d01
```

Reply:
You're right. I messed that up